static/js/components/SearchTextbox.js

```diff
@@ -8,24 +8,49 @@ type Props = {
   value: string
 }
 
-const SearchTextbox = ({ onChange, onClear, onSubmit, value }: Props) => (
-  <div className="search-textbox">
-    <i className="material-icons search-icon" onClick={onSubmit}>
-      search
-    </i>
-    {value ? (
-      <i className="material-icons clear-icon" onClick={onClear}>
-        clear
-      </i>
-    ) : null}
-    <input
-      type="text"
-      className="underlined"
-      value={value}
-      onChange={onChange}
-      onKeyDown={event => (event.key === "Enter" ? onSubmit(event) : null)}
-    />
-  </div>
-)
+export default class SearchTextbox extends React.Component<Props> {
+  input: { current: null | React$ElementRef<typeof HTMLInputElement> }
 
-export default SearchTextbox
+  constructor(props: Props) {
+    super(props)
+    this.input = React.createRef()
+  }
+
+  focus() {
+    if (this.input.current) {
+      this.input.current.focus()
+    }
+  }
+
+  componentDidMount() {
+    this.focus()
+  }
+
+  componentDidUpdate() {
+    this.focus()
+  }
+
+  render() {
+    const { onSubmit, onClear, onChange, value } = this.props
+    return (
+      <div className="search-textbox">
+        <i className="material-icons search-icon" onClick={onSubmit}>
+          search
+        </i>
+        {value ? (
+          <i className="material-icons clear-icon" onClick={onClear}>
+            clear
+          </i>
+        ) : null}
+        <input
+          ref={this.input}
+          type="text"
+          className="underlined"
+          value={value}
+          onChange={onChange}
+          onKeyDown={event => (event.key === "Enter" ? onSubmit(event) : null)}
```

Comment:
just one quick comment - you may be able to do this by just setting the `autofocus` prop on this `<input>` tag

Reply:
Using `autoFocus` will work initially, but if I click the X to clear out the search text, it doesn't retain focus.  The current approach will maintain the focus after clearing the text.

Reply:
ahh gotcha :)